### PR TITLE
Oscar/xor neural network

### PR DESCRIPTION
**Pull request by Eldolfin**
Le reseau de neronne Xor apprends. 
Maintenant je me demande si la demo pourrait pas un peu être améliorée:
pour l'instant on peut la lancer avec l'executable du module NeuralNetwork avec:
**./main -x** ou **./main --xor-demo** et il va afficher un tableau à la fin de l'apprentissage avec les résultats (accuracy et tests passés)
Mais le réseau de neronnes n'est pas sauvegardé. Et on ne peut pas non plus le tester manuellement en entrant des nombres.

Pour amélioré l'executable, faudrait que je suis parser plus d'arguments (genre des paramètres type **./main -x epochs=300 learning_rate=5**) pour le rentre plus interactif / convainquant. Pour ca il faudrait écrire tout un truc pour parser des paramètres ca a l'air long a faire.

Ou alors, une partie pourrait être ajoutée au GUI, qui montrerais une progression au cours de l'apprentissage?

### CODE REVIEW

**Review by Gregoire-perso — COMMENTED** [review]
J'ai regardé vite fait les fichiers de neural network mais sans plus, je te fais confiance sur le fait que ça marche. 
Juste pour le Makefile de test je vois que tu l'as modifié, est ce que c'est normal ? Si non, si tu peux annuler les changements comme ça y'a que les fichiers vraiment modifiés qui seront dans la PR 😄
